tests/integration/test_active_response/test_execd/test_execd_restart.py

```diff
@@ -0,0 +1,168 @@
+        'PROTOCOL': PROTOCOL
+    } for _ in range(0, len(test_metadata))
+]
+
+test_data_path = os.path.join(os.path.dirname(os.path.realpath(__file__)),'data')
+configurations_path = os.path.join(test_data_path, 'wazuh_conf.yaml')
+configurations = load_wazuh_configurations(configurations_path, __name__, params=params, metadata=test_metadata)
+
+remoted_simulator = None
+
+@pytest.fixture(scope="function")
+def start_agent(request, get_configuration):
+    metadata = get_configuration['metadata']
+    authd_simulator = AuthdSimulator(server_address=SERVER_ADDRESS,
+                                     enrollment_port=1515,
+                                     key_path=SERVER_KEY_PATH,
+                                     cert_path=SERVER_CERT_PATH)
+    authd_simulator.start()
+    global remoted_simulator
+    remoted_simulator = RemotedSimulator(server_address=SERVER_ADDRESS,
+                                         remoted_port=1514,
+                                         protocol=PROTOCOL,
+                                         mode='CONTROLED_ACK',
+                                         start_on_init=True,
+                                         client_keys=CLIENT_KEYS_PATH)
+
+    remoted_simulator.setActiveResponseMessage(build_message(metadata, metadata['results']))
+
+    # Clean client.keys file
+    truncate_file(CLIENT_KEYS_PATH)
+    time.sleep(1)
+
+    control_service('stop')
+    agent_auth_pat = 'bin' if platform.system() == 'Linux' else ''
+    subprocess.call([f'{WAZUH_PATH}/{agent_auth_pat}/agent-auth', '-m',
+                    SERVER_ADDRESS])
+    control_service('start')
+
+    yield
+
+    remoted_simulator.stop()
+    authd_simulator.shutdown()
+
+@pytest.fixture(scope="module", params=configurations)
+def get_configuration(request):
+    """Get configurations from the module"""
+    yield request.param
+
+def wait_message_line(line):
+    if platform.system() == 'Windows' and "active-response/bin/restart-wazuh.exe: {\"version\"" in line:
+        return True
+    elif ("ossec/active-response/bin/restart-wazuh: {\"version\"" in line):
+        return True
+    return None
+
+def wait_invalid_input_message_line(line):
+    if ("Invalid input format" in line):
+        return line
+    return None
+
+def wait_shutdown_message_line(line):
+    if ("Shutdown received. Deleting responses." in line):
+        return True
+    return None
+
+def build_message(metadata, expected):
+    origin = "\"name\":\"\",\"module\":\"wazuh-analysisd\""
+    rules = "\"level\":5,\"description\":\"Test.\",\"id\":" + metadata['rule_id']
+
+    if expected['success'] == False:
+        return "{\"origin\":{" + origin + "},\"command\":\"" + metadata['command'] + "\",\"parameters\":{\"extra_args\":[],\"alert\":{\"rule\":{" + rules + "}}}}"
+
+    return "{\"version\":1,\"origin\":{" + origin + "},\"command\":\"" + metadata['command'] + "\",\"parameters\":{\"extra_args\":[],\"alert\":{\"rule\":{" + rules + "}}}}"
```

Comment:
I highly recommend you to using `fstrings` here. Also, to avoid escaping all the `"` characters, just create the string as:
```python
return f'{"version":1...'
```

Reply:
Python complains because the string is too large.

Reply:
You can split the string as I mentioned here https://github.com/wazuh/wazuh-qa/pull/1029#discussion_r571089757